### PR TITLE
DON-1025: Expose mandate next payment date to frontend

### PR DESCRIPTION
**Pull request by bdsl**
Diff in test shows the change in output here:

```diff
              "schedule": {
                    "type": "monthly",
                    "dayOfMonth": 12,
                    "activeFrom": null
+                   "expectedNextPaymentDate": "2024-09-12"
              },
```

### CODE REVIEW

File: src/Domain/RegularGivingService.php [hunk 1]
```diff
@@ -2,21 +2,25 @@
...
+        private RegularGivingMandateRepository $regularGivingMandateRepository,
+        private \DateTimeImmutable $now
+    ) {
     }
 
     public function allActiveForDonorAsApiModel(PersonId $donor): array
     {
         $mandatesWithCharities = $this->regularGivingMandateRepository->allActiveForDonorWithCharities($donor);
 
+        $currentUKTime = $this->now->setTimezone(new \DateTimeZone("Europe/London"));
```
Comment (bdsl) [inline src/Domain/RegularGivingService.php:18]: not sure if this is the best place to do the conversion to UK time.

Comment (NoelLH) [reply src/Domain/RegularGivingService.php:18]: I think while it only happens once + with no variation this is good. I guess until we have charities in other countries, there's not enough value to consider varying times of collection & display TZ by donor – they will likely assume we're using UK ones.

**Review by NoelLH — APPROVED** [review]
LGTM

File: src/Domain/RegularGivingService.php (hunk 1 above)
Comment (bdsl) [reply src/Domain/RegularGivingService.php:18]: Yep. I think even if we have charities in other countries as long as the site is clearly UK based for something like this it'll be fine to have everyone use UK. For something monthly I don't expect many people will mind if it's off up to 12ish hours one way or the other.

**Pull request merged**